tests/api-controller-acceptance-tests/apicontroller/component_test.go

```diff
@@ -0,0 +1,397 @@
+
+	kymaClient := kyma.NewForConfigOrDie(kubeConfig)
+	istioNetClient := istioNet.NewForConfigOrDie(kubeConfig)
+	istioAuthClient := istioAuth.NewForConfigOrDie(kubeConfig)
+
+	Convey("API Controller should", t, func() {
+
+		Convey("create API with authentication disabled", func() {
+			t.Log("create API with authentication disabled")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
+
+			lastApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, lastApi, t, false)
+			So(err, ShouldBeNil)
+			So(lastApi, ShouldNotBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+
+			lastApi, err = ctx.awaitApiCreated(kymaClient, lastApi, true, false)
+			So(err, ShouldBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+			So(lastApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+
+			lastVs, err := istioNetClient.NetworkingV1alpha3().VirtualServices(namespace).Get(lastApi.Status.VirtualServiceStatus.Resource.Name, metav1.GetOptions{})
+			expectedVs := ctx.virtualServiceFor(testId, domainName)
+			So(err, ShouldBeNil)
+			So(lastVs.Spec, ctx.ShouldDeepEqual, expectedVs)
+		})
+
+		Convey("create API with hostname without domain", func() {
+			t.Log("create API with hostname without domain")
+
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, false)
+
+			lastApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, lastApi, t, false)
+			So(err, ShouldBeNil)
+			So(lastApi, ShouldNotBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+
+			lastApi, err = ctx.awaitApiCreated(kymaClient, lastApi, true, false)
+			So(err, ShouldBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+			So(lastApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+
+			lastVs, err := istioNetClient.NetworkingV1alpha3().VirtualServices(namespace).Get(lastApi.Status.VirtualServiceStatus.Resource.Name, metav1.GetOptions{})
+			expectedVs := ctx.virtualServiceFor(testId, domainName)
+			So(err, ShouldBeNil)
+			So(lastVs.Spec, ctx.ShouldDeepEqual, expectedVs)
+		})
+
+		Convey("not create API with wrong domain", func() {
+			t.Log("not create API with wrong domain")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName+"x", apiSecurityDisabled, true)
+
+			_, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("create API with default jwt configuration to enable authentication", func() {
+			t.Log("create API with default jwt configuration to enable authentication")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
```

Comment:
Why do you create api with parameter apiSecurityDisabled and then manually set AuthenticationEnabled to true below? Wouldn't that make more sense to just use an apiSecurityEnabled parameter (deleted by you) instead?